packages/cli/src/lib/implementation/filter.model.ts

```diff
@@ -0,0 +1,27 @@
+import type { GlobalOptions } from '@code-pushup/core';
+import type {
+  CategoryConfig,
+  CoreConfig,
+  PluginConfig,
+} from '@code-pushup/models';
+
+export type FilterCliOptions = {
+  skipCategories?: string[];
+  onlyCategories?: string[];
+  skipPlugins?: string[];
+  onlyPlugins?: string[];
+};
+export type FilterOptions = Partial<GlobalOptions> &
+  Pick<CoreConfig, 'categories' | 'plugins'> &
+  FilterCliOptions;
+
+export type FilterOptionType =
+  | 'skipCategories'
+  | 'onlyCategories'
+  | 'skipPlugins'
+  | 'onlyPlugins';
```

Comment:
`FilterCliOptions` and `FilterOptionType` could derived from each other, makes it easier to keep in sync (you'd only need to change or add an option in one place). Either direction would work:

`FilterCliOptions` :arrow_right: `FilterOptionType`:

```ts
export type FilterCliOptions = {
  skipCategories?: string[];
  onlyCategories?: string[];
  skipPlugins?: string[];
  onlyPlugins?: string[];
};

export type FilterOptionType = keyof FilterCliOptions;
```

`FilterOptionType` :arrow_right: `FilterCliOptions`:

```ts
export type FilterCliOptions = Partial<Record<FilterOptionType, string[]>>;

export type FilterOptionType =
  | 'skipCategories'
  | 'onlyCategories'
  | 'skipPlugins'
  | 'onlyPlugins';
```